### PR TITLE
feat(Components): new select menus

### PR DESCRIPTION
**Pull request by advaith1**
**Please describe the changes this PR makes and why it should be merged:**

new selects! poggers thanks Ian

**If applicable, please reference Discord API Docs PRs or commits that influenced this PR:**

* https://github.com/discord/discord-api-docs/pull/5543
* https://github.com/discord/discord-api-docs/pull/5544

### CODE REVIEW

File: deno/payloads/v10/channel.ts [hunk 1]
```diff
@@ -1306,13 +1306,29 @@ export enum ComponentType {
 	 */
 	Button,
 	/**
-	 * Select Menu component
+	 * Select menu for picking from defined options
 	 */
-	SelectMenu,
```
Comment (vladfrangu) [inline deno/payloads/v10/channel.ts:1311]: Please add this as a deprecated value (see channeltype enum for example)

**Review by vladfrangu — CHANGES_REQUESTED** [review]

File: deno/payloads/v10/channel.ts [hunk 2]
```diff
@@ -1306,13 +1306,29 @@ export enum ComponentType {
...
+	StringSelect,
```
Comment (Jiralite) [inline deno/payloads/v10/channel.ts:1311]: Is this not `TextSelect`?

Comment (advaith1) [reply deno/payloads/v10/channel.ts:1311]: no, docs were fixed in https://github.com/discord/discord-api-docs/pull/5544

**Review by almeidx — CHANGES_REQUESTED** [review]

File: payloads/v10/channel.ts [hunk 3]
```diff
@@ -1329,6 +1329,15 @@ export enum ComponentType {
 	 * Select menu for channels
 	 */
 	ChannelSelect,
+
+	// EVERYTHING BELOW THIS LINE SHOULD BE OLD NAMES FOR RENAMED ENUM MEMBERS //
+
+	/**
+	 * Select menu for picking from defined options
+	 *
+	 * @deprecated This is the old name for {@apilink ChannelType#StringSelect}
```
[the remark this answers is not in the document]
Comment (advaith1) [reply payloads/v10/channel.ts:1338]: whoops

File: deno/payloads/v10/_interactions/base.ts [hunk 4]
```diff
@@ -129,3 +129,36 @@ export type APIGuildInteractionWrapper<Original extends APIBaseInteraction<Inter
...
+	roles?: Record<Snowflake, APIRole>;
+	members?: Record<Snowflake, APIInteractionDataResolvedGuildMember>;
+	channels?: Record<Snowflake, APIInteractionDataResolvedChannel>;
+	attachments?: Record<Snowflake, APIAttachment>;
+}
+
+/**
+ * `users` and optional `members` from APIInteractionDataResolved, for user commands and user selects
+ */
+export type APIUserInteractionDataResolved = Required<Pick<APIInteractionDataResolved, 'users'>> &
```
Comment (vladfrangu) [inline deno/payloads/v10/_interactions/base.ts:163]: The name changes make this PR a major change, can you please add deprecated aliases to the old names? 🙏

File: deno/payloads/v10/channel.ts [hunk 5]
```diff
@@ -1404,15 +1429,26 @@ export enum TextInputStyle {
...
 	/**
-	 * The choices in the select, max 25
+	 * Specified choices in a select menu (only required and available for StringSelect, max 25)
+	 */
+	options?: APISelectMenuOption[];
+	/**
+	 * List of channel types to include in the ChannelSelect component
 	 */
-	options: APISelectMenuOption[];
+	channel_types?: ChannelType[];
```
Comment (vladfrangu) [inline deno/payloads/v10/channel.ts:1451]: You could always make separate interfaces for these based on the types 👀

**Review by vladfrangu — CHANGES_REQUESTED** [review]

**Comment by advaith1** [comment]
should anything else be changed/added here?

**Review by vladfrangu — APPROVED** [review]

**Pull request merged**